### PR TITLE
Update Snackbar actionOverflowThreshold Documentation

### PR DESCRIPTION
**Pull request by RamonFarizel**
This PR addresses an inconsistency in the snackbar widget documentation regarding the percentage threshold for action overflow. The current documentation states that setting the threshold to 0 will prevent the action from overflowing to a new line. However, this behavior is incorrect, as the action still wraps when the threshold is set to 0.

To resolve this, we propose updating the documentation to clarify that the threshold should be set to 1 to ensure the action remains on the same line as the content.

#156817

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/material/snack_bar.dart [hunk 1]
```diff
@@ -426,7 +426,7 @@ class SnackBar extends StatefulWidget {
   /// than this percentage of the width of the snackbar less the width of its [action],
   /// then the [action] will appear below the [content].
   ///
-  /// At a value of 0, the action will not overflow to a new line.
+  /// At a value of 1, the action will not overflow to a new line.
```
Comment (bleroux) [inline packages/flutter/lib/src/material/snack_bar.dart:429]: This does not seem to match the description above.
Maybe this comment was "At a value of 0, the action will always overflow to a new line."

Can you check with various cases? For instance with a content which is way longer than the snackbar width? In that case, even with a value of 1, the condition described in the existing comment should still stand, no?
" /// If the width of the snackbar's [content] is greater
  /// than this percentage of the width of the snackbar less the width of its [action],
  /// then the [action] will appear below the [content]."

Maybe the whole comment is wrong. We should check the bahavior and have a look at the implementation.

Comment (RamonFarizel) [reply packages/flutter/lib/src/material/snack_bar.dart]: @bleroux  I was exploring using 0 and 1 while updating the view width and this was the result:

`actionOverflowThreshold = 0`
https://github.com/user-attachments/assets/7d1e36b7-fbcf-4020-9d97-272eac5bfd4a

`actionOverflowThreshold = 1`
https://github.com/user-attachments/assets/5f70a33b-37c4-428d-9a7b-e53e23768dcf


I do believe "At a value of 0, the action will always overflow to a new line." is the best phrase.

Comment (RamonFarizel) [reply packages/flutter/lib/src/material/snack_bar.dart]: @bleroux 
I double-checked the behavior when the value is set to 1, this time testing with different content sizes, and I found a mistake. I believe this is an edge case, but even when the value is 1, the action can still overflow to a new line.

That said. Can we keep the original comment and just update the phrase about the 0 value?
Like:

```
/// If the width of the snackbar's [content] is greater
/// than this percentage of the width of the snackbar less the width of its [action],
/// then the [action] will appear below the [content].
///
/// At a value of 0, the action will always overflow to a new line.
```


 
Screenshot 2024-10-17 at 18 53 42

Comment (bleroux) [reply packages/flutter/lib/src/material/snack_bar.dart]: Interesting, reading the code it is possible to get a percentage greater than 1 when the action width is greater that the snackbar width (that seems to be the case on your screenshot, where action label is 'Action Action Action'.

I would propose to go with the following:

```dart
/// If the width of the snackbar's [action] divided by the width of the snackbar
/// is greater than this percentage then the [action] will appear below the [content].
///
/// At a value of 0, the action will always overflow to a new line. 
```

Unless your more comfortable with the original comment.

**Review by bleroux — APPROVED** [review]
LGTM!

Thanks for your contribution and taking time to experiment with this behavior.
I proposed to update the original comment but up to you to decide if it is worth it.

File: packages/flutter/lib/src/material/snack_bar.dart (hunk 1 above)
Comment (RamonFarizel) [reply packages/flutter/lib/src/material/snack_bar.dart]: I updated it to follow your suggestion!

**Comment by OutdatedGuy** [comment]
@bleroux @RamonFarizel I was just trying to understand how the overflow is calculated and it's super confusing.

Even though the newly updated docs says that:
```dart
/// If the width of the snackbar's [action] divided by the width of the snackbar
/// is greater than this percentage then the [action] will appear below the [content].
``` 
the action's width used is not the actual size of the `action`. It's the width of text inside the action button (i.e. label's width) + right side margin of the action button which is 12 by default is not specified otherwise.

Also the snackbar's width for `fixed` behaviour is calculated as:

```dart
MediaQuery.sizeOf(context).width - (margin.left + margin.right)
```

but `margin` is only available and used for behaviour of `floating` type.

File: packages/flutter/lib/src/material/snack_bar.dart [hunk 2]
```diff
@@ -422,11 +422,11 @@ class SnackBar extends StatefulWidget {
   /// (optional) The percentage threshold for action widget's width before it overflows
   /// to a new line.
   ///
-  /// Must be between 0 and 1. If the width of the snackbar's [content] is greater
-  /// than this percentage of the width of the snackbar less the width of its [action],
-  /// then the [action] will appear below the [content].
+  /// Must be between 0 and 1.
+  /// If the width of the snackbar's [action] divided by the width of the snackbar
+  /// is greater than this percentage then the [action] will appear below the [content].
```
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/snack_bar.dart:427]: Optional wording suggestion:

```suggestion
  /// If the width of the [action] divided by the total snackbar width
  /// is greater than this percentage, the [action] will appear below the [content].
```

File: packages/flutter/lib/src/material/snack_bar.dart [hunk 3]
```diff
@@ -422,11 +422,11 @@ class SnackBar extends StatefulWidget {
...
   ///
-  /// At a value of 0, the action will not overflow to a new line.
+  /// At a value of 0, the action will always overflow to a new line.
```
Comment (nate-thegrate) [inline packages/flutter/lib/src/material/snack_bar.dart:429]: Just went to the [SnackBar example](https://main-api.flutter.dev/flutter/material/SnackBar-class.html#material.SnackBar.1) and added `actionOverflowThreshold: 0`

Sure enough, a value of zero always causes an overflow!

**Review by nate-thegrate — APPROVED** [review]
LGTM, with optional nit :)

Thanks for catching this documentation mistake and fixing it!

**Comment by nate-thegrate** [comment]
> Also the snackbar's width for `fixed` behaviour is calculated as:
> 
> ```dart
> MediaQuery.sizeOf(context).width - (margin.left + margin.right)
> ```
> 
> but `margin` is only available and used for behaviour of `floating` type.

_oh no_

@OutdatedGuy thanks for bringing this up—I think it's outside the scope of this PR, but we should definitely improve the logic here.

**Pull request merged**